client/client.go

```diff
@@ -247,76 +216,154 @@ func (c *Client) update(latestRoot bool) (data.TargetFiles, error) {
 	if err := c.local.SetMeta("snapshot.json", snapshotJSON); err != nil {
```

Comment:
Acording to [spec](https://theupdateframework.github.io/specification/v1.0.19/index.html#update-snapshot), this should be done right before going to targets.